client/webpack.config.js

```diff
@@ -176,14 +187,16 @@ const filePaths = {
 const webpackConfig = {
 	bail: ! isDevelopment,
 	context: __dirname,
-	entry: filterEntrypoints( {
-		'entry-main': [ path.join( __dirname, 'boot', 'app' ) ],
-		'entry-domains-landing': [ path.join( __dirname, 'landing', 'domains' ) ],
-		'entry-login': [ path.join( __dirname, 'landing', 'login' ) ],
-		'entry-stepper': [ path.join( __dirname, 'landing', 'stepper' ) ],
-		'entry-browsehappy': [ path.join( __dirname, 'landing', 'browsehappy' ) ],
-		'entry-subscriptions': [ path.join( __dirname, 'landing', 'subscriptions' ) ],
-	} ),
+	entry: addHotMiddlewareIfDev(
+		filterEntrypoints( {
```

Comment:
`addHotMiddlewareIfDev` could be part of `filterEntrypoints()` since it's already looping through all the enabled ones anyway.

Reply:
I first tried combining the loops like this:

```diff
diff --git a/client/webpack.config.js b/client/webpack.config.js                                                                        
index a9a964d377..7f814371e8 100644                                                                                                     
--- a/client/webpack.config.js                                                                                                          
+++ b/client/webpack.config.js                                                                                                          
@@ -107,21 +107,18 @@ function filterEntrypoints( entrypoints ) {                                                                       
        const allowed = {};
        Object.entries( entrypoints ).forEach( ( [ key, val ] ) => {
                if ( validEntrypoints.includes( key ) ) {
-                       allowed[ key ] = val;                                                                                           
+                       allowed[ key ] = addHotMiddlewareIfDev( val );                                                                  
                }
        } );

        return allowed;
 }

-function addHotMiddlewareIfDev( entry ) {                                                                                              
-       if ( isDevelopment ) {                                                                                                          
-               return Object.entries( entry ).reduce( ( acc, [ key, value ] ) => {                                                     
-                       acc[ key ] = [ 'webpack-hot-middleware/client', ...value ];                                                     
-                       return acc;                                                                                                     
-               }, {} );                                                                                                                
+function addHotMiddlewareIfDev( value ) {                                                                                              
+       if ( ! isDevelopment ) {                                                                                                        
+               return value;                                                                                                           
        }
-       return entry;                                                                                                                   
+       return [ 'webpack-hot-middleware/client', ...value ];                                                                           
 }

 /**
@@ -187,16 +184,14 @@ const filePaths = {                                                                                               
 const webpackConfig = {
        bail: ! isDevelopment,
        context: __dirname,
-       entry: addHotMiddlewareIfDev(                                                                                                   
-               filterEntrypoints( {                                                                                                    
-                       'entry-main': [ path.join( __dirname, 'boot', 'app' ) ],                                                        
-                       'entry-domains-landing': [ path.join( __dirname, 'landing', 'domains' ) ],                                      
-                       'entry-login': [ path.join( __dirname, 'landing', 'login' ) ],                                                  
-                       'entry-stepper': [ path.join( __dirname, 'landing', 'stepper' ) ],                                              
-                       'entry-browsehappy': [ path.join( __dirname, 'landing', 'browsehappy' ) ],                                      
-                       'entry-subscriptions': [ path.join( __dirname, 'landing', 'subscriptions' ) ],                                  
-               } )                                                                                                                     
-       ),                                                                                                                              
+       entry: filterEntrypoints( {                                                                                                     
+               'entry-main': [ path.join( __dirname, 'boot', 'app' ) ],                                                                
+               'entry-domains-landing': [ path.join( __dirname, 'landing', 'domains' ) ],                                              
+               'entry-login': [ path.join( __dirname, 'landing', 'login' ) ],                                                          
+               'entry-stepper': [ path.join( __dirname, 'landing', 'stepper' ) ],                                                      
+               'entry-browsehappy': [ path.join( __dirname, 'landing', 'browsehappy' ) ],                                              
+               'entry-subscriptions': [ path.join( __dirname, 'landing', 'subscriptions' ) ],                                          
+       } ),                                                                                                                            
        mode: isDevelopment ? 'development' : 'production',
        devtool: sourceMapType,
        output: {
```

Which works when there are entry points set.
However, I discovered the filterEntrypoints also has an early exit case in which it will need the loop anyway:

```
diff --git a/client/webpack.config.js b/client/webpack.config.js
index 7f814371e8..7550883665 100644
--- a/client/webpack.config.js
+++ b/client/webpack.config.js
@@ -83,7 +83,12 @@ if ( shouldCreateSentryRelease ) {
 
 function filterEntrypoints( entrypoints ) {
        if ( ! process.env.ENTRY_LIMIT ) {
-               return entrypoints;
+               return Object.fromEntries(
+                       Object.entries( entrypoints ).map( ( [ key, value ] ) => [
+                               key,
+                               addHotMiddlewareIfDev( value ),
+                       ] )
+               );
        }
 
        const allowedEntrypoints = process.env.ENTRY_LIMIT.split( ',' );
```

Which seems like we really didn't get the benefit of combining the loops, and the logic for adding the middleware is now repeated in multiple spots in the function